src/main/java/ch/njol/skript/bukkitutil/BlockDataUtils.java

```diff
@@ -0,0 +1,104 @@
+package ch.njol.skript.bukkitutil;
+
+import org.bukkit.Material;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+public class BlockDataUtils {
+
+	/**
+	 * Returns a valid block Material based on a given ItemStack, or null if one doesn't exist.
+	 *
+	 * @param item the item to convert
+	 * @return the corresponding block Material, or null if one doesn't exist
+	 */
+	public static @Nullable Material toBlock(ItemStack item) {
+		Material material = item.getType();
+		if (material.isBlock()) // already a block
+			return material;
+
+		return switch (material) {
```

Comment:
Do you know how this will work with multipart blocks (e.g. beds, doors, etc.)?

Reply:
Did some basic tests and it seems to work as intended (i.e. returning the proper material)